packages/react/src/components/Forms/FileUploader/index.tsx

```diff
@@ -0,0 +1,347 @@
+           */
+          if (!container.current!.contains(e.relatedTarget as Node)) {
+            setDragActive(false);
+          }
+        }}
+        onDragOver={(e) => {
+          e.preventDefault();
+        }}
+        onDrop={(e) => {
+          // To prevent a new tab to open.
+          e.preventDefault();
+          const newFiles = Array.from(e.dataTransfer.files);
+          if (inputRef.current) {
+            inputRef.current.files = e.dataTransfer.files;
+            onFilesChange([...newFiles]);
+          }
+          setDragActive(false);
+        }}
+        ref={container}
+      >
+        <div className="c__file-uploader__inner">
+          {renderIcon()}
+          {children || (
+            <>
+              <div className="c__file-uploader__inner__caption">
+                {renderCaption()}
+              </div>
+              {text && (
+                <div className="c__file-uploader__inner__text">{text}</div>
+              )}
+            </>
+          )}
+          <input
+            type="file"
+            name={name}
+            ref={inputRef}
+            onChange={(e) => {
+              if (e.target.files) {
+                onFilesChange([...e.target.files]);
+              } else {
+                onFilesChange([]);
+              }
+            }}
+            multiple={multiple}
+            {...inputProps}
+          />
+        </div>
+      </label>
+    );
+  }
+);
+
+const FileUploaderMono = forwardRef<FileUploaderRefType, FileUploaderProps>(
+  ({ fakeDefaultFiles, ...props }, ref) => {
+    const { t } = useCunningham();
+    const [file, setFile] = useState<File | undefined>(
+      fakeDefaultFiles && fakeDefaultFiles.length > 0
+        ? fakeDefaultFiles[0]
+        : undefined
+    );
+    // This is made to prevent useEffects inside DropZone that depends on `files` to trigger on each render,
+    // doing this preserves the reference of the array.
+    const files = useMemo(() => (file ? [file] : []), [file]);
+    const [hoverDelete, setHoverDelete] = useState(false);
+    const [icon, setIcon] = useState<ReactNode>();
+    const [animateIcon, setAnimateIcon] = useState(true);
+
+    const deleteFile = (e: React.MouseEvent<HTMLElement>) => {
+      setFile(undefined);
+      // This is to prevent opening the browse file window.
+      e.preventDefault();
+      setHoverDelete(false);
+    };
+
+    const updateIcon = () => {
+      if (hoverDelete) {
+        setAnimateIcon(true);
+        setIcon(
+          props.deleteIcon ?? <span className="material-icons">delete</span>
+        );
+        return;
+      }
+      if (file) {
+        setAnimateIcon(false);
+        setIcon(
+          props.fileSelectedIcon ?? (
+            <span className="material-icons">download</span>
+          )
+        );
+        return;
+      }
+      setAnimateIcon(true);
+      setIcon(props.icon);
+    };
+
+    useEffect(() => {
+      updateIcon();
+    }, [file, hoverDelete]);
```

Comment:
I feel we can remove extra rerender by removing this `useEffect` and `animationIcon` and `icon` states and process animateIcon and icon directly within the render or through a `useMemo`

Reply:
yes good idea